### PR TITLE
[Refactor]Merged string_value into string_ref

### PR DESCRIPTION
**Pull request by zclllyybb**
## Problem summary

merged these two classes with the same purpose and usages.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

Initial work is done for cr. There need & will be new commits soon to make it run.

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.27 seconds
 load time: 511 seconds
 storage size: 17123157108 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230117115143_clickbench_pr_82129.html

File: be/src/exprs/new_agg_fn_evaluator.cc [hunk 1]
```diff
@@ -255,8 +255,8 @@ void NewAggFnEvaluator::SetDstSlot(const AnyVal* src, const SlotDescriptor& dst_
     case TYPE_OBJECT:
     case TYPE_QUANTILE_STATE:
     case TYPE_STRING:
-        *reinterpret_cast<StringValue*>(slot) =
-                StringValue::from_string_val(*reinterpret_cast<const StringVal*>(src));
+        //
```
Comment (HappenLee) [inline be/src/exprs/new_agg_fn_evaluator.cc:258]: remove the useless code

File: be/src/vec/common/hash_table/string_hash_table.h [hunk 2]
```diff
@@ -373,8 +377,6 @@ class StringHashTable : private boost::noncopyable {
                 }
                 }
             }
-            while (need_switch_to_next)
```
Comment (HappenLee) [inline be/src/vec/common/hash_table/string_hash_table.h:376]: what the code mean ？ rethink to delete code is safe？

File: be/src/olap/in_list_predicate.h [hunk 3]
```diff
@@ -120,15 +114,16 @@ class InListPredicateBase : public ColumnPredicate {
         if constexpr (is_string_type(Type) || Type == TYPE_DECIMALV2 || is_date_type(Type)) {
             _values = new phmap::flat_hash_set<T>();
             auto values = ((HybridSetType*)hybrid_set.get())->get_inner_set();
+            // type = phmap::flat_hash_set<std::string>
```
[the remark this answers is not in the document]
Comment (zclllyybb) [reply be/src/olap/in_list_predicate.h:117]: it's type hint. I'll make it clearer

File: be/src/vec/common/hash_table/string_hash_table.h (hunk 2 above)
Comment (zclllyybb) [reply be/src/vec/common/hash_table/string_hash_table.h:376]: double checked it. should write in LOG(FATAL, ...). I'll move it to another pr.

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**